pkg/apis/sources/v1/ping_validation.go

```diff
@@ -94,6 +94,7 @@ func (cs *PingSourceSpec) Validate(ctx context.Context) *apis.FieldError {
 			}
 		}
 	}
+	errs = errs.Also(cs.SourceSpec.Validate(ctx))
```

Comment:
Can we do this for all other sources as well?

Reply:
Sure, but, I don't know exactly where that should be. If you can give me a hand with this :)

Reply:
Sure, we need to go through every file in `pkg/apis/sources/<version>/*_validation.go`, and check if the embedded `SourceSpec` is validated, if not then we add the same check that you added for the `PingSource` (line 97).

For  example, for `ApiServerSource`, we don't validate `SourceSpec` [1], so we need to add the check `errs = errs.Also(cs.SourceSpec.Validate(ctx))` here https://github.com/knative/eventing/blob/407f3cbe8fabdab04894631b7b0c61bed083bccb/pkg/apis/sources/v1/apiserver_validation.go#L75

Also, if we can add simple unit tests for each source to prevent regression that would be great too.

Hope this helps, thanks!

[1]
https://github.com/knative/eventing/blob/407f3cbe8fabdab04894631b7b0c61bed083bccb/pkg/apis/sources/v1/apiserver_validation.go#L39-L77